### PR TITLE
TTL quick setup phase 2

### LINKED ISSUE #1255: Timed traffic lights on T-junction for pedestrians always green
_Originally posted by @krzychu124 in https://github.com/CitiesSkylinesMods/TMPE/issues/1241#issuecomment-1001260420_

> @kianzarrin Have you seen that report? Maybe you could fit a bugfix in the PR (it looks like tiny bug somewhere in auto-tl code)? https://steamcommunity.com/workshop/filedetails/discussion/1637663252/3200368477539744008/
 
Steam user wrote:

> When setting up a timed traffic light (no matter if manual or with ctrl+click) the traffic lights for the pedestrians on the top left of the T-junction remain always green, regardless of which other lights are showing red or green. They count down green from 7 to 0 and directly restart with 7 for green.

### PR DESCRIPTION
**Pull request by kianzarrin**
[TMPE.zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=alt-TTL)
TTL quick setup for + intersection with dedicated far turn cycles. #575
image

ban crossing when far segment does not have near turn. fixes #1255.
image

### CODE REVIEW

File: TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs [hunk 1]
```diff
@@ -136,11 +136,16 @@ public TimedTrafficLightsTool(TrafficManagerTool mainTool)
...
+            if (Shortcuts.ControlIsPressed) {
+                AutoTimedTrafficLights.ErrorResult res;
+                if (Shortcuts.AltIsPressed) {
+                    res = AutoTimedTrafficLights.SetupSingleFar(HoveredNodeId);
+                } else {
+                    res = AutoTimedTrafficLights.Setup(HoveredNodeId);
+                }
                 string message = null;
 
+                Log._Debug(res.ToString());
```
Comment (kvakvs) [inline TLM/TLM/UI/SubTools/TTL/TimedTrafficLightsTool.cs:148]: Either log this with explanation useful for support and log reading, or remove debug log

File: TLM/TLM/Util/AutoTimedTrafficLights.cs [hunk 2]
```diff
@@ -89,6 +91,97 @@ public enum ErrorResult {
...
+            if ((flags & NetNode.Flags.LevelCrossing) != 0) {
+                Log._Debug("not level crossing");
+                return ErrorResult.NotSupported;
+            }
+
+            var segList = ArrangedSegments(nodeId);
+            int n = segList.Count;
+
+            if (n != 4) {
+                Log._Debug("n=" + n);
```
Comment (kvakvs) [inline TLM/TLM/Util/AutoTimedTrafficLights.cs:114]: Change log message to something useful, or remove debug log

File: TLM/TLM/Util/AutoTimedTrafficLights.cs [hunk 3]
```diff
@@ -89,6 +91,97 @@ public enum ErrorResult {
...
+                return ErrorResult.NotSupported;
+            }
+
+            foreach(ushort segmentId in segList) {
+                if (ExtSegmentManager.Instance.CalculateIsOneWay(segmentId)) {
+                    Log._Debug("oneway");
```
Comment (kvakvs) [inline TLM/TLM/Util/AutoTimedTrafficLights.cs:120]: Remove debug log

**Review by kvakvs — APPROVED** [review]
Code looks good only some debug logs should be removed or converted to real logs

**Review by krzychu124 — APPROVED** [review]
Looks good, I haven't had a chance to try in-game but the code looks ok so I don't expect issues there

**Comment by kianzarrin** [comment]
One more problem: I do need dedicated right turn. So I should not use the single left traffic lights mode

**Comment by kianzarrin** [comment]
OK fixed 
image

**Comment by kvakvs** [comment]
Tested Alt+Ctrl+Click new mode on
* 3 way T-junction: did not work
* 4 way junction: looks good, 4 states: road 1 straight+near, all far turn, road 2 straight+near, then all far turn again
* 5 way junction: did not work (usual ctrl+click works)

Suggestion: Mention in the tooltip (onscreen display) that only works for 4 way junctions

**Comment by kianzarrin** [comment]
> Suggestion: Mention in the tooltip (onscreen display) that only works for 4 way junctions

we can do that in the translation right?

**Comment by kvakvs** [comment]
> > Suggestion: Mention in the tooltip (onscreen display) that only works for 4 way junctions
> 
> we can do that in the translation right?

Yes totally. Write that in english translation and the language translators will pick it up to their languages.

**Pull request merged**